frigate/detectors/plugins/memryx.py

```diff
@@ -0,0 +1,600 @@
+                    f"Post-processing model found at {post_model_file_path}, skipping download."
+                )
+            else:
+                logger.info(
+                    f"Model files not found. Downloading from {self.model_url}..."
+                )
+                zip_path = os.path.join(self.cache_dir, "memryx_model.zip")
+
+                # Download the ZIP file
+                urllib.request.urlretrieve(self.model_url, zip_path)
+                logger.info(f"Model ZIP downloaded to {zip_path}. Extracting...")
+
+                # Extract ZIP file
+                with zipfile.ZipFile(zip_path, "r") as zip_ref:
+                    zip_ref.extractall(self.cache_dir)
+
+                logger.info(f"Model extracted to {self.cache_dir}.")
+
+                # Assign extracted files to correct paths
+                for file in os.listdir(self.cache_dir):
+                    if file == self.expected_post_model:
+                        self.memx_post_model = os.path.join(self.cache_dir, file)
+
+                logger.info(f"Assigned Model Path: {self.memx_model_path}")
+                logger.info(
+                    f"Assigned Post-processing Model Path: {self.memx_post_model}"
+                )
+
+                # Cleanup: Remove the ZIP file after extraction
+                os.remove(zip_path)
+                logger.info("Cleaned up ZIP file after extraction.")
+
+    def send_input(self, connection_id, tensor_input: np.ndarray):
+        """Pre-process (if needed) and send frame to MemryX input queue"""
+        if tensor_input is None:
+            raise ValueError("[send_input] No image data provided for inference")
+
+        if self.memx_model_type == ModelTypeEnum.yolox:
+            tensor_input = tensor_input.squeeze(0)
+
+            padded_img = np.ones((640, 640, 3), dtype=np.uint8) * 114
+
+            scale = min(
+                640 / float(tensor_input.shape[0]), 640 / float(tensor_input.shape[1])
+            )
+            sx, sy = (
+                int(tensor_input.shape[1] * scale),
+                int(tensor_input.shape[0] * scale),
+            )
+
+            resized_img = cv2.resize(
+                tensor_input, (sx, sy), interpolation=cv2.INTER_LINEAR
+            )
+            padded_img[:sy, :sx] = resized_img.astype(np.uint8)
+
+            # Step 4: Slice the padded image into 4 quadrants and concatenate them into 12 channels
+            x0 = padded_img[0::2, 0::2, :]  # Top-left
+            x1 = padded_img[1::2, 0::2, :]  # Bottom-left
+            x2 = padded_img[0::2, 1::2, :]  # Top-right
+            x3 = padded_img[1::2, 1::2, :]  # Bottom-right
+
+            # Step 5: Concatenate along the channel dimension (axis 2)
+            concatenated_img = np.concatenate([x0, x1, x2, x3], axis=2)
+            processed_input = concatenated_img.astype(np.float32)
+        else:
+            processed_input = tensor_input.astype(np.float32) / 255.0  # Normalize
```

Comment:
similar to below this is already supported as the model config `input_dtype` we should use that, potentially adding a new `float32_norm` type. That way this else branch won't be needed at all.

Reply:
The main `float` input_dtype is actually already normalized